crates/sui-json-rpc/src/read_api.rs

```diff
@@ -316,54 +319,95 @@ impl ReadApi {
 
         if opts.show_events {
             trace!("getting events");
-
-            let event_digests_list = temp_response
+            let events_digests_list = temp_response
                 .values()
                 .filter_map(|cache_entry| match &cache_entry.effects {
                     Some(eff) => eff.events_digest().cloned(),
                     None => None,
                 })
                 .collect::<Vec<TransactionEventsDigest>>();
+            // filter out empty events digest, as they do not have to be read from the DB
+            let empty_events_digest = TransactionEvents::default().digest();
+            let events_digests_list = events_digests_list
+                .into_iter()
+                .filter(|d| d != &empty_events_digest)
+                .collect::<Vec<_>>();
 
-            // fetch events from the DB
-            let events = self
-                .transaction_kv_store
-                .multi_get_events(&event_digests_list)
+            let mut events_digest_to_events = if events_digests_list.is_empty() {
+                HashMap::new()
+            } else {
+                // fetch events from the DB with retry
+                let backoff = ExponentialBackoff {
+                    max_elapsed_time: Some(Duration::from_secs(30)),
+                    ..ExponentialBackoff::default()
+                };
```

Comment:
I'm very torn on this. I don't really think its a good idea to have read queries like this be able to hang for 30 additional seconds before returning.

Reply:
it's not ideal indeed, do you suggest a lower timeout?

Reply:
yes, or maybe even just retry a single extra time as a best-effort

Reply:
makes sense, lowered the timeout to 3 seconds, and the backoff multiplier to 1 for faster retry and err out.